build.zig

```diff
@@ -187,22 +187,24 @@ fn build_zkvm_targets(b: *Builder, main_exe: *Builder.Step) !void {
     });
     zeam_types.addImport("@zeam/params", zeam_params);
 
-    // add state transition
-    const zeam_state_transition = b.addModule("@zeam/state-transition", .{
-        .root_source_file = b.path("pkgs/state-transition/src/lib.zig"),
-        .target = target,
-        .optimize = optimize,
-    });
-    zeam_state_transition.addImport("@zeam/types", zeam_types);
-    zeam_state_transition.addImport("ssz", ssz);
-
     for (zkvm_targets) |zkvm_target| {
         const zkvm_module = b.addModule("zkvm", .{
             .optimize = optimize,
             .target = target,
             .root_source_file = b.path(b.fmt("pkgs/state-transition-runtime/src/{s}/lib.zig", .{zkvm_target.name})),
         });
 
+        // add state transition, create a new module for each zkvm since
+        // that module depends on the zkvm module.
+        const zeam_state_transition = b.addModule("@zeam/state-transition", .{
```

Comment:
can't we only add the `zkvm` addImport to this module inside the loop and else keep it outside the loop?

Reply:
no because the module itself is different for each zkvm binary, the import is a reference to the module so we can't do that afaik.